kv/kvserver/batcheval: TestEvalAddSSTable failed

kv/kvserver/batcheval.TestEvalAddSSTable [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4306835&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4306835&tab=artifacts#/) on master @ [678e51ec2fd4885374a821fb7fdef82fd4d865b1](https://github.com/cockroachdb/cockroach/commits/678e51ec2fd4885374a821fb7fdef82fd4d865b1):


```
=== RUN   TestEvalAddSSTable
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable2087432818
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvalAddSSTable
    cmd_add_sstable_test.go:766: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable2087432818
--- FAIL: TestEvalAddSSTable (6.28s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
initialized metamorphic constant "addsst-rewrite-forced" with value true
    cmd_add_sstable_test.go:753: 
        	Error Trace:	cmd_add_sstable_test.go:753
        	Error:      	Not equal: 
        	            	expected: []sstutil.KV{sstutil.KV{KeyString:"a", WallTimestamp:7, ValueString:"a7"}, sstutil.KV{KeyString:"a", WallTimestamp:6, ValueString:"a6"}}
        	            	actual  : []sstutil.KV{sstutil.KV{KeyString:"a", WallTimestamp:8, ValueString:"a7"}, sstutil.KV{KeyString:"a", WallTimestamp:6, ValueString:"a6"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -3,3 +3,3 @@
        	            	   KeyString: (string) (len=1) "a",
        	            	-  WallTimestamp: (int64) 7,
        	            	+  WallTimestamp: (int64) 8,
        	            	   ValueString: (string) (len=2) "a7"
        	Test:       	TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.03s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=false
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=false (3.27s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
initialized metamorphic constant "addsst-rewrite-forced" with value true
    cmd_add_sstable_test.go:753: 
        	Error Trace:	cmd_add_sstable_test.go:753
        	Error:      	Not equal: 
        	            	expected: []sstutil.KV{sstutil.KV{KeyString:"a", WallTimestamp:7, ValueString:"a7"}, sstutil.KV{KeyString:"a", WallTimestamp:6, ValueString:"a6"}}
        	            	actual  : []sstutil.KV{sstutil.KV{KeyString:"a", WallTimestamp:8, ValueString:"a7"}, sstutil.KV{KeyString:"a", WallTimestamp:6, ValueString:"a6"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -3,3 +3,3 @@
        	            	   KeyString: (string) (len=1) "a",
        	            	-  WallTimestamp: (int64) 7,
        	            	+  WallTimestamp: (int64) 8,
        	            	   ValueString: (string) (len=2) "a7"
        	Test:       	TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.02s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=true (2.97s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvalAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestEvalAddSSTable [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4306920&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4306920&tab=artifacts#/) on master @ [270dca1002697a721316104906f02ec8cce7e7a2](https://github.com/cockroachdb/cockroach/commits/270dca1002697a721316104906f02ec8cce7e7a2):


```
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (6) key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.03s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=false
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=false (4.01s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
initialized metamorphic constant "addsst-rewrite-forced" with value true
    cmd_add_sstable_test.go:703: 
        	Error Trace:	cmd_add_sstable_test.go:703
        	Error:      	Received unexpected error:
        	            	updating SST timestamps: rewriting data blocks: key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:86
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | testing.tRunner
        	            	  | 	/usr/local/go/src/testing/testing.go:1259
        	            	Wraps: (2) updating SST timestamps
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteKeySuffixesInBlocks
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:67
        	            	  | github.com/cockroachdb/pebble/sstable.RewriteKeySuffixes
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:35
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.UpdateSSTTimestamps
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/storage/sst.go:237
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:84
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | [...repeated from below...]
        	            	Wraps: (4) rewriting data blocks
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteBlocks
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:153
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteDataBlocksToWriter.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:214
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (6) key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.05s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=true (3.39s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvalAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestEvalAddSSTable [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4307013&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4307013&tab=artifacts#/) on master @ [678e51ec2fd4885374a821fb7fdef82fd4d865b1](https://github.com/cockroachdb/cockroach/commits/678e51ec2fd4885374a821fb7fdef82fd4d865b1):


```
=== RUN   TestEvalAddSSTable
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable3714956531
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvalAddSSTable
    cmd_add_sstable_test.go:766: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable3714956531
--- FAIL: TestEvalAddSSTable (6.81s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
initialized metamorphic constant "addsst-rewrite-forced" with value true
    cmd_add_sstable_test.go:753: 
        	Error Trace:	cmd_add_sstable_test.go:753
        	Error:      	Not equal: 
        	            	expected: []sstutil.KV{sstutil.KV{KeyString:"a", WallTimestamp:7, ValueString:"a7"}, sstutil.KV{KeyString:"a", WallTimestamp:6, ValueString:"a6"}}
        	            	actual  : []sstutil.KV{sstutil.KV{KeyString:"a", WallTimestamp:8, ValueString:"a7"}, sstutil.KV{KeyString:"a", WallTimestamp:6, ValueString:"a6"}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -3,3 +3,3 @@
        	            	   KeyString: (string) (len=1) "a",
        	            	-  WallTimestamp: (int64) 7,
        	            	+  WallTimestamp: (int64) 8,
        	            	   ValueString: (string) (len=2) "a7"
        	Test:       	TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.05s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=true (2.87s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvalAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestEvalAddSSTable [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4307187&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4307187&tab=artifacts#/) on master @ [270dca1002697a721316104906f02ec8cce7e7a2](https://github.com/cockroachdb/cockroach/commits/270dca1002697a721316104906f02ec8cce7e7a2):


```
=== RUN   TestEvalAddSSTable
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable1538438007
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvalAddSSTable
    cmd_add_sstable_test.go:766: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable1538438007
--- FAIL: TestEvalAddSSTable (6.52s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=false
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=false (3.56s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
initialized metamorphic constant "addsst-rewrite-forced" with value true
    cmd_add_sstable_test.go:703: 
        	Error Trace:	cmd_add_sstable_test.go:703
        	Error:      	Received unexpected error:
        	            	updating SST timestamps: rewriting data blocks: key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:86
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | testing.tRunner
        	            	  | 	/usr/local/go/src/testing/testing.go:1259
        	            	Wraps: (2) updating SST timestamps
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteKeySuffixesInBlocks
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:67
        	            	  | github.com/cockroachdb/pebble/sstable.RewriteKeySuffixes
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:35
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.UpdateSSTTimestamps
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/storage/sst.go:237
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:84
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | [...repeated from below...]
        	            	Wraps: (4) rewriting data blocks
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteBlocks
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:153
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteDataBlocksToWriter.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:214
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (6) key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.03s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvalAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestEvalAddSSTable [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4307465&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4307465&tab=artifacts#/) on master @ [270dca1002697a721316104906f02ec8cce7e7a2](https://github.com/cockroachdb/cockroach/commits/270dca1002697a721316104906f02ec8cce7e7a2):


```
=== RUN   TestEvalAddSSTable
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable2923819469
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvalAddSSTable
    cmd_add_sstable_test.go:766: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestEvalAddSSTable2923819469
--- FAIL: TestEvalAddSSTable (6.26s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
initialized metamorphic constant "addsst-rewrite-forced" with value true
    cmd_add_sstable_test.go:703: 
        	Error Trace:	cmd_add_sstable_test.go:703
        	Error:      	Received unexpected error:
        	            	updating SST timestamps: rewriting data blocks: key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:86
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | testing.tRunner
        	            	  | 	/usr/local/go/src/testing/testing.go:1259
        	            	Wraps: (2) updating SST timestamps
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteKeySuffixesInBlocks
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:67
        	            	  | github.com/cockroachdb/pebble/sstable.RewriteKeySuffixes
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:35
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.UpdateSSTTimestamps
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/storage/sst.go:237
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:84
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | [...repeated from below...]
        	            	Wraps: (4) rewriting data blocks
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteBlocks
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:153
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteDataBlocksToWriter.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/pebble/sstable/suffix_rewriter.go:214
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (6) key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.04s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=true (2.79s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvalAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestEvalAddSSTable [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4308242&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4308242&tab=artifacts#/) on master @ [6f4c69b0b3fae65eabb3c053313db7e1d6c8aeda](https://github.com/cockroachdb/cockroach/commits/6f4c69b0b3fae65eabb3c053313db7e1d6c8aeda):


```
=== RUN   TestEvalAddSSTable
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestEvalAddSSTable1154423246
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvalAddSSTable
    cmd_add_sstable_test.go:766: -- test log scope end --
--- FAIL: TestEvalAddSSTable (3.63s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
    cmd_add_sstable_test.go:703: 
        	Error Trace:	cmd_add_sstable_test.go:703
        	Error:      	Received unexpected error:
        	            	updating SST timestamps: rewriting data blocks: key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:86
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1259
        	            	Wraps: (2) updating SST timestamps
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteKeySuffixesInBlocks
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:67
        	            	  | github.com/cockroachdb/pebble/sstable.RewriteKeySuffixes
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:35
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.UpdateSSTTimestamps
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/sst.go:237
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:84
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | [...repeated from below...]
        	            	Wraps: (4) rewriting data blocks
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteBlocks
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:153
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteDataBlocksToWriter.func1
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:214
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (6) key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=false/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.02s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=false
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=false (1.93s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvalAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/batcheval.TestEvalAddSSTable [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4308243&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4308243&tab=artifacts#/) on master @ [6f4c69b0b3fae65eabb3c053313db7e1d6c8aeda](https://github.com/cockroachdb/cockroach/commits/6f4c69b0b3fae65eabb3c053313db7e1d6c8aeda):


```
=== RUN   TestEvalAddSSTable
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/8b7adba2afa4d2ee9e34e55bcb54994d/logTestEvalAddSSTable1733437817
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestEvalAddSSTable
    cmd_add_sstable_test.go:766: -- test log scope end --
--- FAIL: TestEvalAddSSTable (4.25s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
    cmd_add_sstable_test.go:703: 
        	Error Trace:	cmd_add_sstable_test.go:703
        	Error:      	Received unexpected error:
        	            	updating SST timestamps: rewriting data blocks: key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:86
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1259
        	            	Wraps: (2) updating SST timestamps
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteKeySuffixesInBlocks
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:67
        	            	  | github.com/cockroachdb/pebble/sstable.RewriteKeySuffixes
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:35
        	            	  | github.com/cockroachdb/cockroach/pkg/storage.UpdateSSTTimestamps
        	            	  | 	github.com/cockroachdb/cockroach/pkg/storage/sst.go:237
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval.EvalAddSSTable
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval/cmd_add_sstable.go:84
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test.TestEvalAddSSTable.func1.1
        	            	  | 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/batcheval_test/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go:667
        	            	  | [...repeated from below...]
        	            	Wraps: (4) rewriting data blocks
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteBlocks
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:153
        	            	  | github.com/cockroachdb/pebble/sstable.rewriteDataBlocksToWriter.func1
        	            	  | 	github.com/cockroachdb/pebble/sstable/external/com_github_cockroachdb_pebble/sstable/suffix_rewriter.go:214
        	            	  | runtime.goexit
        	            	  | 	GOROOT/src/runtime/asm_amd64.s:1581
        	            	Wraps: (6) key has suffix "\x00\x00\x00\x00\x00\x00\x00\a\t", expected "\x00\x00\x00\x00\x00\x00\x00\b\t"
        	            	Error types: (1) *withstack.withStack (2) *errutil.withPrefix (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.leafError
        	Test:       	TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race
        --- FAIL: TestEvalAddSSTable/IngestAsWrites=true/SSTTimestamp_doesn't_rewrite_with_incorrect_timestamp,_but_errors_under_race (0.02s)
=== RUN   TestEvalAddSSTable/IngestAsWrites=true
    --- FAIL: TestEvalAddSSTable/IngestAsWrites=true (1.96s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEvalAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go
+++ b/pkg/kv/kvserver/batcheval/cmd_add_sstable_test.go
@@ -622,6 +622,8 @@ func TestEvalAddSSTable(t *testing.T) {
 			expect:         []sstutil.KV{{"a", 7, "a7"}, {"a", 6, "a6"}},
 			expectStatsEst: true,
 		},
+		/* Disabled due to nondeterminism under metamorphic tests. SSTTimestamp will
+		 * shortly be removed anyway.
 		"SSTTimestamp doesn't rewrite with incorrect timestamp, but errors under race": {
 			atReqTS:        8,
 			data:           []sstutil.KV{{"a", 6, "a6"}},
@@ -630,7 +632,7 @@ func TestEvalAddSSTable(t *testing.T) {
 			expect:         []sstutil.KV{{"a", 7, "a7"}, {"a", 6, "a6"}},
 			expectErrRace:  `incorrect timestamp 0.000000007,0 for SST key "a" (expected 0.000000008,0)`,
 			expectStatsEst: true,
-		},
+		},*/
 	}
 	testutils.RunTrueAndFalse(t, "IngestAsWrites", func(t *testing.T, ingestAsWrites bool) {
 		for name, tc := range testcases {

